enhancement: preview for text messages

andersonjeccel opened the pull request:
| Q                                      | A
| -------------------------------------- | --------------------------
| Bug fix?          | ❌
| New feature/enhancement?      | ✔️
| Deprecations?                          | ❌
| BC breaks?        | ❌
| Automated tests included?              | ❌
| Related user documentation PR URL      | 
| Related developer documentation PR URL | 
| Issue(s) addressed                     | 

## Description

This PR redesigns preview functionality for text messages in Mautic. Users can now see how their SMS messages will appear on mobile devices through a realistic phone mockup interface.

This enhancement significantly improves the user experience when creating and editing SMS campaigns by providing immediate visual feedback on how messages will appear to recipients.

<img width="1250" height="875" alt="image" src="https://github.com/user-attachments/assets/4ea9d0e0-4c7d-4a1f-85b4-0649f7a63636" />


---

### 📋 Steps to test this PR:

1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Navigate to Channels > Text Messages
3. Create a new SMS message or edit an existing one
4. Go to the SMS details page
5. Look for the phone preview on the right side of the page
6. Verify the preview shows your message content in a realistic phone interface
7. Check that the preview scales properly on different screen sizes
8. Verify the demo message and input placeholder text are properly translated

andersonjeccel committed:
Final preview template

andersonjeccel committed:
Add to details page

andersonjeccel committed:
Darker shadow box

andersonjeccel committed:
Improve layout responsiveness

andersonjeccel committed:
Add slide animation

andersonjeccel commented:
@escopecz I'd appreciate some guidance on where to keep the CSS styles, since this bundle has no custom CSS file 
Adding it could require using a function to import on details.html.twig and preview.html.twig _(I'm not sure if this is a clean approach)_

kuzmany reviewed (approved):
it works nicely

kuzmany commented on app/bundles/SmsBundle/Resources/views/Sms/preview.html.twig:
```diff
@@ -1,4 +1,477 @@
+</div>
+
+<style>
+.phone-preview__button--outer {
```
I recommend put css to app/bundles/SmsBundle/Assets/css/sms.css

andersonjeccel replied:
Oh, creating the file makes it get automatically loaded
Thanks!

andersonjeccel committed:
Moving CSS to file

matbcvo commented:
@andersonjeccel, could you add `word-break` or something like that to handle long word?"

<img width="372" height="509" alt="image" src="https://github.com/user-attachments/assets/ae220319-86bc-4bd9-a50b-60d89eb2d162" />

kuzmany committed:
Fix text overflow in SMS preview components

- Add word-break CSS properties to phone-preview input field
- Add word-break CSS properties to phone-preview message containers
- Add word-break CSS properties to phone-preview message content
- Prevents layout issues with long words/text in SMS preview

Addresses text overflow concerns in mobile phone preview interface.

kuzmany commented:
@matbcvo now should be better

<img width="695" height="702" alt="image" src="https://github.com/user-attachments/assets/9ebc450b-ee3d-48d2-a627-1d90eb6da138" />

matbcvo committed:
Merge branch '7.x' into enhancement/sms-preview
